### ISSUE TITLE
[skrifa] Paths equivalent but representation different for स in Noto Sans Devanagari

### ISSUE THREAD

**Issue by drott** [googlefonts/fontations#570]
With [this Skia CL #740176](https://skia-review.googlesource.com/c/skia/+/740176) applied and viewer built with the following command line:
`# bazel build -c dbg -with_fontations //tools/viewer`

I get these differing outputs:
[Diff here](https://gist.github.com/drott/82a0f0f98911080e69efbbb74c649ce7)

I have not yet tried to draw the two results, however, opening the glyph in Samsa, it looks like it has overlapping the components:

<img width="583" alt="image" src="https://github.com/googlefonts/fontations/assets/1021784/5bac2566-d319-4bc4-8fb2-1d420d0cfae3">

which makes me wonder whether it has something to do with FreeType performing processing on them. 

Visually, comparing the two paths with path ops, they appear to be equivalent in terms of the area that they cover.

```
Different path for glyph index: 0 glyph id: 60, font: fonts/noto/NotoSansDevanagari-Regular.ttf data sizes 3195 vs 2673
Path A (Fontations): 
path.setFillType(SkPathFillType::kWinding);
path.moveTo(5.8828125f, -13.2246094f);
path.lineTo(6.55078125f, -13.5820312f);
path.quadTo(6.81445312f, -13.2011719f, 6.984375f, -12.5507812f);
path.quadTo(7.15429688f, -11.90625f, 7.15429688f, -10.9921875f);
path.quadTo(7.15429688f, -9.9375f, 6.90234375f, -9.15820312f);
path.quadTo(6.65039062f, -8.37890625f, 6.19335938f, -7.86914062f);
path.lineTo(6.16992188f, -7.65820312f);
path.quadTo(5.73632812f, -7.1015625f, 5.14453125f, -6.69140625f);
path.quadTo(4.55859375f, -6.28710938f, 3.81445312f, -6);
path.lineTo(3.9140625f, -6.21679688f);
path.quadTo(4.8984375f, -4.9453125f, 6.25195312f, -3.57421875f);
path.quadTo(7.60546875f, -2.20898438f, 9.1171875f, -1.0078125f);
path.lineTo(7.82226562f, 0.287109375f);
path.quadTo(6.48046875f, -0.890625f, 5.29101562f, -2.0859375f);
path.quadTo(4.1015625f, -3.28710938f, 3.1875f, -4.34179688f);
path.quadTo(2.27929688f, -5.40234375f, 1.75195312f, -6.14648438f);
path.quadTo(1.22460938f, -6.890625f, 1.08984375f, -7.31835938f);
path.quadTo(0.9609375f, -7.75195312f, 0.9609375f, -8.16210938f);
path.quadTo(0.9609375f, -8.61328125f, 1.31835938f, -9.0234375f);
path.quadTo(1.68164062f, -9.43359375f, 2.49609375f, -9.43359375f);
path.quadTo(3.09375f, -9.43359375f, 3.5390625f, -9.15820312f);
path.quadTo(3.984375f, -8.8828125f, 4.34179688f, -8.390625f);
path.quadTo(4.70507812f, -7.8984375f, 5.015625f, -7.20117188f);
path.lineTo(3.55078125f, -7.9453125f);
path.quadTo(4.22460938f, -8.13867188f, 4.5703125f, -8.61328125f);
path.quadTo(4.921875f, -9.09375f, 5.0625f, -9.7265625f);
path.quadTo(5.20898438f, -10.3652344f, 5.20898438f, -10.9921875f);
path.quadTo(5.20898438f, -11.90625f, 5.0390625f, -12.5625f);
path.quadTo(4.86914062f, -13.2246094f, 4.65820312f, -13.5351562f);
path.lineTo(5.68945312f, -13.2246094f);
path.lineTo(0, -13.2246094f);
path.lineTo(0, -14.9296875f);
path.lineTo(10.0078125f, -14.9296875f);
path.lineTo(10.0078125f, -13.2246094f);
path.lineTo(5.8828125f, -13.2246094f);
path.close();
path.moveTo(14.0859375f, -13.2246094f);
path.lineTo(14.0859375f, 0);
path.lineTo(12.1464844f, 0);
path.lineTo(12.1464844f, -13.2246094f);
path.lineTo(9.50390625f, -13.2246094f);
path.lineTo(9.50390625f, -14.9296875f);
path.lineTo(16.5820312f, -14.9296875f);
path.lineTo(16.5820312f, -13.2246094f);
path.lineTo(14.0859375f, -13.2246094f);
path.close();
path.moveTo(8.85351562f, -7.72851562f);
path.quadTo(9.67382812f, -7.72851562f, 10.3417969f, -7.79882812f);
path.quadTo(11.015625f, -7.86914062f, 11.6835938f, -8.02734375f);
path.quadTo(12.3574219f, -8.18554688f, 13.1308594f, -8.47265625f);
path.lineTo(13.1308594f, -6.72070312f);
path.quadTo(12.09375f, -6.31054688f, 11.2441406f, -6.1640625f);
path.quadTo(10.3945312f, -6.0234375f, 9.1171875f, -6.0234375f);
path.quadTo(8.32617188f, -6.0234375f, 7.44726562f, -6.12890625f);
path.quadTo(6.57421875f, -6.24023438f, 5.80664062f, -6.4453125f);
path.quadTo(5.0390625f, -6.65039062f, 4.55859375f, -6.890625f);
path.lineTo(5.0390625f, -8.32617188f);
path.quadTo(5.51953125f, -8.16210938f, 6.140625f, -8.015625f);
path.quadTo(6.76757812f, -7.86914062f, 7.46484375f, -7.79882812f);
path.quadTo(8.16210938f, -7.72851562f, 8.85351562f, -7.72851562f);
path.close();

Path B (FreeType): 
path.setFillType(SkPathFillType::kWinding);
path.moveTo(6.75114536f, -13.2246094f);
path.lineTo(12.1464844f, -13.2246094f);
path.lineTo(12.1464844f, -8.14787674f);
path.quadTo(11.9094486f, -8.0803709f, 11.6835938f, -8.02734375f);
path.quadTo(11.015625f, -7.86914062f, 10.3417969f, -7.79882812f);
path.quadTo(9.67382812f, -7.72851562f, 8.85351562f, -7.72851562f);
path.quadTo(8.16210938f, -7.72851562f, 7.46484375f, -7.79882812f);
path.quadTo(6.84980536f, -7.86084843f, 6.28947687f, -7.98213482f);
path.quadTo(6.67754078f, -8.462883f, 6.90234375f, -9.15820312f);
path.quadTo(7.15429688f, -9.9375f, 7.15429688f, -10.9921875f);
path.quadTo(7.15429688f, -11.90625f, 6.984375f, -12.5507812f);
path.quadTo(6.88408852f, -12.9346371f, 6.75114536f, -13.2246094f);
path.close();
path.moveTo(9.50390625f, -14.9296875f);
path.lineTo(0, -14.9296875f);
path.lineTo(0, -13.2246094f);
path.lineTo(4.82290888f, -13.2246094f);
path.quadTo(4.93821001f, -12.9554768f, 5.0390625f, -12.5625f);
path.quadTo(5.20898438f, -11.90625f, 5.20898438f, -10.9921875f);
path.quadTo(5.20898438f, -10.3652344f, 5.0625f, -9.7265625f);
path.quadTo(4.921875f, -9.09375f, 4.5703125f, -8.61328125f);
path.quadTo(4.47500658f, -8.48243713f, 4.35476208f, -8.37296963f);
path.quadTo(4.3482995f, -8.38181496f, 4.34179688f, -8.390625f);
path.quadTo(3.984375f, -8.8828125f, 3.5390625f, -9.15820312f);
path.quadTo(3.09375f, -9.43359375f, 2.49609375f, -9.43359375f);
path.quadTo(1.68164062f, -9.43359375f, 1.31835938f, -9.0234375f);
path.quadTo(0.9609375f, -8.61328125f, 0.9609375f, -8.16210938f);
path.quadTo(0.9609375f, -7.75195312f, 1.08984375f, -7.31835938f);
path.quadTo(1.22460938f, -6.890625f, 1.75195312f, -6.14648438f);
path.quadTo(2.27929688f, -5.40234375f, 3.1875f, -4.34179688f);
path.quadTo(4.1015625f, -3.28710938f, 5.29101562f, -2.0859375f);
path.quadTo(6.48046875f, -0.890625f, 7.82226562f, 0.287109375f);
path.lineTo(9.1171875f, -1.0078125f);
path.quadTo(7.60546875f, -2.20898438f, 6.25195312f, -3.57421875f);
path.quadTo(4.96983957f, -4.87298298f, 4.01894951f, -6.08236742f);
path.quadTo(4.61422729f, -6.33253384f, 5.10198641f, -6.66234589f);
path.quadTo(5.41739988f, -6.54930782f, 5.80664062f, -6.4453125f);
path.quadTo(6.57421875f, -6.24023438f, 7.44726562f, -6.12890625f);
path.quadTo(8.32617188f, -6.0234375f, 9.1171875f, -6.0234375f);
path.quadTo(10.3945312f, -6.0234375f, 11.2441406f, -6.1640625f);
path.quadTo(11.6715813f, -6.23775864f, 12.1464844f, -6.37819576f);
path.lineTo(12.1464844f, 0);
path.lineTo(14.0859375f, 0);
path.lineTo(14.0859375f, -13.2246094f);
path.lineTo(16.5820312f, -13.2246094f);
path.lineTo(16.5820312f, -14.9296875f);
path.lineTo(9.50390625f, -14.9296875f);
path.close();
```
Font attached:
[NotoSansDevanagari-Regular.ttf.zip](https://github.com/googlefonts/fontations/files/12315083/NotoSansDevanagari-Regular.ttf.zip)

**Comment by drott** [comment]
@anthrotype found https://skia-review.googlesource.com/c/skia/+/521637 which does seem to tigger running a PathOps Simplify operation on the glyph with overlapping contours.

**Comment by drott** [comment]
Glyph with gid 60 does have that flag set:

```
<TTGlyph name="uni0938" xMin="0" yMin="-12" xMax="691" yMax="622">
      <contour>
        <pt x="245" y="551" on="1" overlap="1"/>
        <pt x="273" y="566" on="1"/>
[...]
```

**Comment by drott** [comment]
Closed, this is addressed in testing by using #581 and simplifying paths.
